### DISCUSSION TITLE
foundry - libusb. WHY?

### DISCUSSION THREAD

**Discussion by kidoMiike** [Cyfrin/foundry-full-course-cu#197] [General]
Hi guys. Could anyone explain why does foundry needs libusb to be installed? What has it to do with usb ports at all? (I am a bit worried, because I have a cold wallet, and now I'm installing some software that needs to have special utilities for usb). 
Thanks for answers.

**Comment by PatrickAlphaC** [comment]
Amazing question, I asked in the foundry support telegram. 

https://t.me/foundry_support/39755

**Reply by kidoMiike** [reply]
Wow! I’ve read the thread! Seems to be a real concern. Thanks for taking the question to the right place. For now I won’t worry that much and believe they are honest devs and cool guys :)

**Comment by gakonst** [comment]
Hey - LibUSB is used in Cast / Forge Script to be able to talk to Ledger and other devices. We haven't been able to pull that dependency out super cleanly, so for now you need it but it's not going to give any unauthorized access to any device.

**Reply by kidoMiike** [reply]
Thanks for the answer, I hope it will be something optional in the future, cause it’s a bit scary now. Program that knows how to generate private keys etc. uses a special utility to connect to cold wallets :) and I have installed it.
